app/scripts/metamask-controller.js

```diff
@@ -4940,6 +4945,7 @@ export default class MetamaskController extends EventEmitter {
         ),
         getAccountType: this.getAccountType.bind(this),
         getDeviceModel: this.getDeviceModel.bind(this),
+        isConfirmationRedesignEnabled,
         snapAndHardwareMessenger: this.controllerMessenger.getRestricted({
```

Comment:
It may be better to inject preference controller state rather than injecting single property.

Reply:
I did initially pass the preferencesController and then we had a conversation to pass the single prop here
https://github.com/MetaMask/metamask-extension/pull/24182#discussion_r1580908661. Open to hearing more opinions on this

Reply:
I initially passed the preferencesController and then we had a conversation to pass the single prop here
https://github.com/MetaMask/metamask-extension/pull/24182#discussion_r1580908661. Open to hearing more opinions on this

Reply:
My thinking was it avoids coupling another entire controller to a specific middleware, plus it allowed us to encapsulate the knowledge of the env in this file, rather than the middleware, and potentially other places in future.